enhancements/fbc-run-bundle.md

```diff
@@ -0,0 +1,268 @@
+- [X] Test plan is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+- Do we need to build a File-Based Catalog to run this command?
+
+  - We can use `quay.io/operatorhubio/catalog:latest` for testing and generate a File-Based Catalog at runtime when the run bundle command is invoked.
+
+- Do we need to make our own image to use as a default one for run bundle and run bundle-upgrade?
+
+  - Currently, we are using `quay.io/operator-framework/opm:latest` as the default index image. We might have to build an image similar to `quay.io/operatorframework/catalog:latest` which uses opm as the base image and also has a built catalog in it.
+
+- While initializing the FBC, should the channel received from the bundle annotations be used to initialize the default channel name in the `opm init` command?
+
+  - We should be able to handle bundles that don't have channel metadata on the annotations.
+
+- If we don’t get channel information from the bundle, then can we add it to a default channel or to the channel HEAD?
+
+- Can we assume the Channel’s “entries” field be initialized to an array with only one element, i.e., {name: “api-operator.v.0.1”} with no “replaces” field?
+
+  - "replaces" field is required for `run bundle-upgrade` but for run bundle, I think it's ok to assume for the channel entries to have just one entry.
+
+## Summary
+
+The goal of this enhancement is to discuss the impact of the new file based catalog (FBC) in Operator Lifecyle Manager and how it affects the Operator SDK, specifically the existing run-bundle command. This EP discusses the implementation/workflow changes of the `run bundle` command to support the new File-Based Catalog (FBC) index format along with continuing to support SQLite based index images.
+
+## Motivation
+
+One of the main goals of the Operator SDK is to enable operator developers to install and run their operators using Operator Lifecycle Manager (OLM). Currently, there are two commands in the Operator SDK for installing an operator bundle (`run bundle`) and upgrading an operator bundle to a newer version (`run bundle-upgrade`) using OLM. Operator SDK installs and upgrades the operator bundles using opm tooling that inherently uses a SQLite database under the hood, which is responsible for generating the registry database as well as the index images. We use `opm registry add` under the hood to create an ephemeral index image during the `run bundle(-upgrade)` workflow; however, that command is being deprecated.
+
+As OLM is transitioning to declarative config which is a more human readable and consumable format of a catalog that is in the form of `JSON/YAML`, Operator SDK should transition the existing `run bundle(-upgrade)` commands to support declarative config index images and default to using file based catalogs so that Operator authors can continue using Operator SDK to install/upgrade their operator bundles using the new FBC format.
```

Comment:
```suggestion
As OLM is transitioning to declarative config, which is a more human readable and consumable format of a catalog that is in the form of `JSON/YAML`, Operator SDK should transition the existing `run bundle(-upgrade)` commands to support declarative config index images and default to using file based catalogs so that Operator authors can continue using Operator SDK to install/upgrade their operator bundles using the new FBC format.
```